File: Backend/routes/details.py
from flask import Blueprint, request, jsonify

from models import Person, Property

details_bp = Blueprint('details', __name__, url_prefix='/api')

@details_bp.route('/details', methods=['GET'])
def get_details():
    item_id = request.args.get('id')
    item_type = request.args.get('type')

    if not item_id or not item_type:
        return jsonify({'error': 'Missing parameters: id and type required'}), 400

    if item_type == 'people':
        person = Person.query.get(item_id)
        if not person:
            return jsonify({'error': 'Person not found'}), 404

        return jsonify({
            'id': person.id,
            'name': person.name,
            'gender': person.gender,
            'city': person.city,
            'looking_for': person.looking_for,
            'hobbies': person.hobbies,
            'description': person.description,
            'preferences': person.preferences,
            'contact': person.contact,
            'profile_picture': getattr(person, 'profile_picture', None),
            # 'government_id': getattr(person, 'government_id', None)
        })

    elif item_type == 'properties':
        prop = Property.query.get(item_id)
        if not prop:
            return jsonify({'error': 'Property not found'}), 404

        return jsonify({
            'id': prop.id,
            'name': prop.name,
            'city': prop.city,
            'address': prop.address,
            'rent': prop.rent,
            'security_amount': float(prop.security_amount),
            'electricity_rate': float(prop.electricity_rate),
            'amenities': prop.amenities,
            'pictures': prop.pictures,
            'contact': prop.contact,
            'gender_preference': getattr(prop, 'gender_preference', 'any'),
            # 'government_id': getattr(prop, 'government_id', None)
        })

    return jsonify({'error': 'Invalid type: must be "people" or "properties"'}), 400

File: Backend/static/JS/details.js
document.addEventListener('DOMContentLoaded', async () => {
    const params = new URLSearchParams(window.location.search);
    const id = params.get("id");
    const type = params.get("type");

    fetch(`/api/details?id=${id}&type=${type}`)
        .then(res => res.json())
        .then(data => {
            if (data.error) {
                document.querySelector('.right-section').innerText = data.error;
                return;
            }

            document.getElementById("name").innerText = data.name || "Unknown";
            document.querySelector("#location span").innerText = data.city || "Unknown City";

            // Image
            let imgUrl = '';

            if (type === 'people') {
                const firstName = data.name?.split(' ')[0] || 'User';
                const picPath = data.profile_picture?.replace(/^uploads\//, '');

                imgUrl = data.profile_picture
                    ? `/uploads/${picPath}`
                    : `https://ui-avatars.com/api/?name=${encodeURIComponent(firstName)}&background=random&bold=true`;

            } else {
                const picPath = data.pictures?.[0]?.replace(/^uploads\//, '');

                imgUrl = data.pictures?.[0]
                    ? `/uploads/${picPath}`
                    : 'https://images.unsplash.com/photo-1600585154340-be6161a56a0c?auto=format&fit=crop&w=600&q=80';
            }


            document.getElementById("image-section").innerHTML = `<img src="${imgUrl}" alt="Preview Image">`;

            // Basic Info
            const basicInfo = document.getElementById("basic-info");
            basicInfo.innerHTML = `
                ${data.gender ? `<span>Gender: ${data.gender}</span>` : ''}
                ${type === 'people' && data.looking_for ? `<span>Looking for: ${data.looking_for}</span>` : ''}
                ${type === 'properties' && data.rent?.single ? `<span>Single Rent: ₹${data.rent.single}</span>` : ''}
                ${data.security_amount ? `<span>Deposit: ₹${data.security_amount}</span>` : ''}
                ${data.electricity_rate ? `<span>Electricity: ₹${data.electricity_rate}/unit</span>` : ''}
            `;

            // Features (Hobbies / Amenities)
            const features = document.getElementById("features");
            const list = type === 'people'
                ? [...(data.hobbies || []), ...Object.values(data.preferences || {})].filter(Boolean)
                : (data.amenities || []);
            features.innerHTML = list.map(f => `<span class="feature">${f}</span>`).join('');

            // Highlights (dummy example)
            const highlights = document.getElementById("highlights");
            if (type === 'people') {
                highlights.innerHTML = `<span class="feature">✔️ Working full time</span>`;
            }

            // Description
            document.getElementById("description").innerText = data.description || "No description provided.";

            // Contact
            const contactBtn = document.getElementById("contact");
            if (data.contact) {
                contactBtn.onclick = () => window.open(`tel:${data.contact}`, '_self');
            } else {
                contactBtn.style.display = 'none';
            }

        })
        .catch(err => {
            console.error(err);
            document.querySelector('.right-section').innerText = "Failed to load data.";
        });
});
